### DISCUSSION TITLE
Exclude files & directories from deployment

### DISCUSSION THREAD

**Discussion by maxim-belkin** [JamesIves/github-pages-deploy-action#633] [Q&A] [answered]
Hi,
I've got a couple of questions:
1. Is it possible to exclude certain files and/or directories that are present in the folder being deployed?
2. Is it possible to  deploy specific folders and/or files (based on patterns) only instead of a single folder?

My understanding is that currently one has to "clean things up" and move everything into one folder before using this action -- is this indeed so?

Thanks in advance for your help!

**Comment by JamesIves** [comment, marked as answer]
You're able to exclude certain files and folders from the deployment job using the clean-exclude parameter. The second is not currently possible, what is your current use case for pattern matching like that?

**Comment by maxim-belkin** [comment]
> The second is not currently possible, what is your current use case for pattern matching like that?

We (The Carpentries, https://github.com/carpentries) develop lessons using either Markdown or Rmarkdown. For lessons developed in RMarkdown, we use GitHub action to convert them to Markdown files, then push them to `gh-pages` branch to let GitHub build and deploy websites, see https://github.com/carpentries/styles/blob/gh-pages/.github/workflows/website.yml for an example.
What I'd like to do is replace the `Commit and Push` step in that action with your action.  Essentially, I want to exclude certain paths from being copied to the `gh-pages` branch

> You're able to exclude certain files and folders from the deployment job using the clean-exclude parameter. 

I saw that parameter but the directories I'd like to exclude are not related to "hashed files" (btw, it's not clear if "hashed files" refer to files used by `hashFiles()` or some other files).

**Reply by JamesIves** [reply]
Hashed files are referring to [content hashed files from Webpack/rollup](https://webpack.js.org/guides/caching/) etc. The clean option removes differences between the deployment branch and the base branch, it prevents build up of unused files between deployment jobs. In hindsight it might not be what you're looking for.

It might make sense to create a simple script that runs before the deployment job that pushes all of the folders you want to deploy into a dist folder, and then deploy that directly. Having an ignore files option doesn't seem particularly useful as your own script could be the decider on that, and that way you have more control over the whole process.

**Reply by sanchom** [reply]
> It might make sense to create a simple script that runs before the deployment job that pushes all of the folders you want to deploy into a dist folder, and then deploy that directly.

That's exactly what I do with another publication system that I use. I prep a staging folder that looks exactly what I want my live gh-pages to look like. Then I deploy that.

**Reply by maxim-belkin** [reply]
Makes sense! Thank you both, @sanchom and @JamesIves. I'll definitely try that.
